pom.xml

```diff
@@ -53,6 +54,28 @@
             <artifactId>mailer</artifactId>
             <!-- TODO: should be transformed into a optional dependency -->
         </dependency>
+        <dependency>
```

Comment:
@basil I think this should be avoided, can you confirm please?

Reply:
On the recipe-update side, I think the plugin stopped building with early iterations of this PR due to lack of `javax.mail` (did newer Jenkins ecosystem stop depending on the module?)

If "avoiding" means not using javax.mail in favor of something else, that refactoring can be a worthwhile separate issue, for someone who knows Java well :)

(I guess similar goes for mockito?)

Reply:
On the recipe-update side, I think the plugin stopped building with early iterations of this PR due to lack of `javax.mail` (did newer Jenkins ecosystem stop depending on the module?)

If "avoiding" means not using javax.mail in favor of something else, that refactoring can be a worthwhile separate issue, for someone who knows Java well (well, better than myself) :)

(I guess similar goes for mockito?)

Reply:
javax has been dropped in favour of jakarta mail, it's the same thing just packages rename. 

It was split from Jenkins core and may need to be changed to jakarta for it to work at some point.

Reply:
So `javax` -> `jakarta` in pom and code should nail it? That bit I can try in the evening :)

Reply:
Simpler just to drop the direct dependency on Java Mail entirely:

```java
diff --git a/src/test/java/hudson/plugins/im/IMPublisherTest.java b/src/test/java/hudson/plugins/im/IMPublisherTest.java
index 350c97e..f318cfe 100644
--- a/src/test/java/hudson/plugins/im/IMPublisherTest.java
+++ b/src/test/java/hudson/plugins/im/IMPublisherTest.java
@@ -25,8 +25,6 @@ import java.util.Iterator;
 import java.util.Map;
 import java.util.Set;
 
-import javax.mail.MessagingException;
-
 import org.junit.Before;
 import org.junit.Test;
 
@@ -149,7 +147,7 @@ public class IMPublisherTest {
      * list.
      */
     @Test
-    public void testIncludeUpstreamCulprits() throws MessagingException, InterruptedException {
+    public void testIncludeUpstreamCulprits() throws Exception {
         Set<User> recipients = this.imPublisher.getNearestUpstreamCommitters(this.build, listener).keySet();
 
         assertEquals(recipients.toString(), 2, recipients.size());
```

That way your only dependency on Java Mail is transitive and you won't have to worry about it in this plugin.

Reply:
Seems to work, thanks!